mongo/options/clientencryptionoptions.go

```diff
@@ -9,18 +9,24 @@ package options
 import (
 	"crypto/tls"
 	"fmt"
+	"net/http"
+
+	"go.mongodb.org/mongo-driver/internal"
 )
 
 // ClientEncryptionOptions represents all possible options used to configure a ClientEncryption instance.
 type ClientEncryptionOptions struct {
 	KeyVaultNamespace string
 	KmsProviders      map[string]map[string]interface{}
 	TLSConfig         map[string]*tls.Config
+	HTTPClient        *http.Client
```

Comment:
Optional: Consider adding a comment that describes what the `HTTPClient` configuration is used for.

E.g.
```go
// HTTPClient allows specifying a custom HTTP client used to fetch credentials from local
// credentials services (e.g. the EC2 instance metadata service). If nil, then the Go Driver
// uses its default HTTP client (internal.DefaultHTTPClient).
```